app/src/main/java/tv/vizbee/demo/activity/MainActivity.kt

```diff
@@ -1,20 +1,55 @@
 package tv.vizbee.demo.activity
 
-import androidx.appcompat.app.AppCompatActivity
+import android.content.Intent
 import android.os.Bundle
 import android.view.Menu
+import android.view.MenuItem
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
 import tv.vizbee.demo.R
 import tv.vizbee.demo.fragments.IFragmentController
+import tv.vizbee.demo.fragments.UserLoginFragment
 import tv.vizbee.demo.fragments.VideoDetailsFragment
 import tv.vizbee.demo.fragments.VideoGalleryFragment
+import tv.vizbee.demo.helper.SharedPreferenceHelper
 import tv.vizbee.demo.model.VideoItem
+import tv.vizbee.demo.retrofit.ApiInterface
+import tv.vizbee.demo.retrofit.RetrofitInstance
+import tv.vizbee.demo.vizbee.VizbeeHomeSSOAdapter
+import tv.vizbee.demo.vizbee.VizbeeWrapper
+import tv.vizbee.utils.Logger
 
 class MainActivity : AppCompatActivity(),
     IFragmentController {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
         showVideoGalleryFragment()
+        handleLogin(intent)
+        Logger.d(LOG_TAG, "handleLogin onCreate")
+
+    }
+
+    override fun onNewIntent(intent: Intent?) {
+        super.onNewIntent(intent)
+        intent?.let {
+            Logger.d(LOG_TAG, "handleLogin onNewIntent")
+            handleLogin(it)
+        }
+    }
+    private fun handleLogin(intent: Intent) {
+        val bundle = intent.extras
+        if (bundle != null) {
```

Comment:
You can directly use 
intent.extras?.let {
  var launchLogin = bundle.getBoolean(VizbeeHomeSSOAdapter.INTENT_LAUNCH_LOGIN_SCREEN)
            if (launchLogin) {
                Logger.d(LOG_TAG, "LoginFragment Launch Called")
                showUserLoginFragment()
            }
}

Reply:
Done